Engine/src/Resources/TextureManager.cpp

```diff
@@ -68,17 +68,14 @@ SharedPtr<Texture> TextureManager::Load(const std::string& textureName, std::uin
     auto it = m_textures.find(textureName);
 
     if (it == m_textures.end()) {
-        switch (Renderer::GetAPI()) {
-        case RendererAPI::API::OpenGL: {
-            SharedPtr<Texture> texture = MakeSharedPtr<OpenGLTexture>(width, height, nrChannels);
-            if (save) {
-                m_textures.insert({ textureName, texture });
-            }
-            return texture;
-        }
-        default:
-            EL_CORE_ASSERT(false, "Unknown Renderer API!");
+        SharedPtr<Texture> texture = Texture::Create(
```

Comment:
Backend selection for texture creation was done inline in texture manager so I had to move it to Texture::Create